models/cart/src/cart/builder.spec.ts

```diff
@@ -1,267 +1,250 @@
-          userRef: expect.objectContaining({ typeId: 'user' }),
-        }),
-        lastModifiedBy: expect.objectContaining({
-          customerRef: expect.objectContaining({ typeId: 'customer' }),
-          userRef: expect.objectContaining({ typeId: 'user' }),
+      ]),
+      totalPrice: expect.objectContaining({
+        centAmount: expect.any(Number),
+        currencyCode: expect.any(String),
+        __typename: 'Money',
+      }),
+      createdBy: expect.objectContaining({
+        clientId: expect.any(String),
+        customerRef: expect.objectContaining({ typeId: 'customer' }),
+        userRef: expect.objectContaining({ typeId: 'user' }),
+        __typename: 'Initiator',
+      }),
+      lastModifiedBy: expect.objectContaining({
+        clientId: expect.any(String),
+        customerRef: expect.objectContaining({ typeId: 'customer' }),
+        userRef: expect.objectContaining({ typeId: 'user' }),
+        __typename: 'Initiator',
+      }),
+      businessUnitRef: expect.objectContaining({
+        typeId: 'business-unit',
+        __typename: 'Reference',
+      }),
+      customerGroupRef: expect.objectContaining({
+        typeId: 'customer-group',
+        __typename: 'Reference',
+      }),
+      refusedGiftsRefs: expect.arrayContaining([
+        expect.objectContaining({
+          typeId: 'cart-discount',
+          __typename: 'Reference',
         }),
-        totalLineItemQuantity: expect.any(Number),
-        shippingInfo: null,
-        directDiscounts: expect.arrayContaining([]),
-        taxedPrice: null,
-        taxedShippingPrice: null,
-        paymentInfo: null,
-        __typename: 'Cart',
-      })
-    )
+      ]),
+      storeRef: expect.objectContaining({
+        typeId: 'store',
+        __typename: 'Reference',
+      }),
+      businessUnit: expect.objectContaining({
+        id: expect.any(String),
+        __typename: 'BusinessUnit',
+      }),
+      customerGroup: expect.objectContaining({
+        id: expect.any(String),
+        __typename: 'CustomerGroup',
+      }),
+      refusedGifts: expect.arrayContaining([
+        expect.objectContaining({
+          id: expect.any(String),
+          __typename: 'CartDiscount',
+        }),
+      ]),
+      store: expect.objectContaining({
+        id: expect.any(String),
+        __typename: 'Store',
+      }),
+    })
   );
+};
+
+describe('Cart model builders', () => {
+  it('builds a REST model', () => {
+    const restModel = populateRestModel(
+      CartRest.random().lineItems([LineItem.random()])
+    ).build();
+
+    validateRestModel(restModel);
+  });
+
+  it('builds a GraphQL model', () => {
+    const graphqlModel = populateGraphqlModel(
+      CartGraphql.random().lineItems([LineItem.random()])
+    ).build();
+
+    validateGraphqlModel(graphqlModel);
+  });
+});
+
+describe('Cart model compatibility builders', () => {
+  it('builds a REST model', () => {
+    const restModel = populateRestModel(Cart.random()).buildRest();
```

Comment:
Should we be populating the `lineItems` here as well to have all tests with the same set of data?